### PR TITLE
RHIDP-3975 Managing authorization using the REST API

### PR DESCRIPTION
**Pull request by themr0c**
* Managing authorization using the REST API
* Moved Understanding permission policy and role configuration source

**IMPORTANT: Do Not Merge - To be merged by Docs Team Only**

**Version(s):** 1.3.1, 1.4

**Issue:** https://issues.redhat.com/browse/RHIDP-3975

**Link to docs preview:**  https://redhat-developer.github.io/red-hat-developers-documentation-rhdh/pr-685/authorization/#managing-authorizations-by-using-the-rest-api

**Reviews:**
- [ ] SME: @ mention assignee
- [ ] QE: @ mention assignee
- [x] Docs review: @ mention assignee
- [ ] Additional review: @mention assignee (by writer)

**Additional information:**

### CODE REVIEW

**Comment by themr0c** [comment]
@AndrienkoAleksandr can you review also

File: modules/authorization/con-permission-policy-and-role-source.adoc [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+
+Managing roles and permission policies originating from CSV files and REST API involves straightforward modification based on their initial source information.
+
+The Configuration file pertains to the default `role:default/rbac_admin` role provided by the RBAC plugin.
+The default role has limited permissions to create, read, update, and delete permission policies or roles, and to read catalog entities.
+
+[NOTE]
+====
+In case the default permissions are insufficient for your administrative requirements, you can create a custom admin role with required permission policies.
```
Comment (AndrienkoAleksandr) [inline modules/authorization/con-permission-policy-and-role-source.adoc:22]: We can also provide information that our RBAC backend supports a super admin option in the configuration (rbac.admin.superUsers).

File: modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+__<endpoint>__::
+Enter the xref:ref-rbac-rest-api-endpoints_{context}[API endpoint] to which you want to send a request, such as `/api/permission/policies`.
+
+__<body>__::
+Enter the JSON body with data that your xref:ref-rbac-rest-api-endpoints_{context}[API endpoint] might need with the HTTP `POST` request.
+
+.Example request to create or update a role
+[source,subs="+attributes,+quotes"]
+----
```
Comment (AndrienkoAleksandr) [inline modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc:54]: With "create" I agree. About "update", I think we have another response body format and url pattern too:

```
curl -v -H "Content-Type: application/json" \
  -H "Authorization: Bearer _<token>_" \
  -X PUT "{my-product-url}/api/permission/roles/role/default/test" \
  -d '{
          "oldRole": {
            "memberReferences":  [ "group:default/example" ],
            "name": "role:default/test",
          },
          "newRole": {
            "memberReferences": [ "group:default/example", "user:default/test" ],
            "name": "role:default/test"
          }
        }'
```

File: modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  -H "Authorization: Bearer _<token>_" \
+  -X POST "{my-product-url}/api/permission/roles" \
+  -d '{
+      "memberReferences": ["group:default/example"],
+      "name": "role:default/test",
+      "metadata": { "description": "This is a test role" }
+    }'
+----
+
+.Example request to create or update a permission policy
```
Comment (AndrienkoAleksandr) [inline modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc:65]: With "create" I agree. About "update", I think we have another response body format and url pattern too:

```
curl -v -H "Content-Type: application/json" \
  -H "Authorization: Bearer $token" \
  -X POST "{my-product-url}/api/permission/policies/role/default/test" \
  -d '{
         "oldPolicy": [
           { 
             "permission": "catalog-entity", "policy": "read", "effect": "allow"
           }
         ],
         "newPolicy":
           [
             { 
               "permission": "policy-entity", "policy": "read", "effect": "allow"
             }
           ]
       }'
```

File: modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+      "permission": "catalog-entity",
+      "policy": "read", "effect":"allow"
+    }]'
+----
+
+.Example request to create or update a condition
+[source,subs="+attributes,+quotes"]
+----
+
+curl -v -H "Content-Type: application/json" \
```
Comment (AndrienkoAleksandr) [inline modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc:82]: I agree with 'create'. Regarding 'update', I believe we have a different URL pattern. The user needs to specify a condition ID. The POST method returns the condition ID, and it is also present in the conditions JSON when the user retrieves the list of conditions. Here is an example of the PUT request:

```
curl -v -H "Content-Type: application/json" \
  -H "Authorization: Bearer $token" \
  -X PUT "{my-product-url}/api/permission/roles/conditions/1" \
  -d '{
          "result":"CONDITIONAL",
          "roleEntityRef":"role:default/test",
          "pluginId":"catalog",
          "resourceType":"catalog-entity",
          "permissionMapping": ["read",  "update", "delete"],
          "conditions": {
            "rule": "IS_ENTITY_OWNER",
            "resourceType": "catalog-entity",
            "params": {"claims": ["group:default/janus-authors"]}
          }
       }'
```

File: modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  -H "Authorization: Bearer $token" \
+  -X POST "{my-product-url}/api/permission/roles/conditions"\
```
Comment (AndrienkoAleksandr) [inline modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc:84]:
```suggestion
  -X POST "{my-product-url}/api/permission/roles/conditions" \
```

**Comment by AndrienkoAleksandr** [comment]
I can also share with you gist API examples: https://gist.githubusercontent.com/AndrienkoAleksandr/d4f383e7e80122681327136e7a2caa9c/raw/c68975e3d4986654cc88fbae441a957705b49800/gistfile1.txt

File: modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+pass:c,a,q[{my-product-url}]::
+Enter your {product-short} URL.
+
+__<endpoint>__::
+Enter the xref:ref-rbac-rest-api-endpoints_{context}[API endpoint] to which you want to send a request, such as `/api/permission/policies`.
+
+__<body>__::
+Enter the JSON body with data that your xref:ref-rbac-rest-api-endpoints_{context}[API endpoint] might need with the HTTP `POST` request.
+
+.Example request to create or update a role
```
Comment (themr0c) [inline modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc:52]:
```suggestion
.Example request to create a role
```

File: modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc (hunk 3 above)
Comment (themr0c) [inline modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc:65]:
```suggestion
.Example request to create a permission policy
```

File: modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+curl -v -H "Content-Type: application/json" \
+  -H "Authorization: Bearer _<token>_" \
+  -X POST "{my-product-url}/api/permission/roles" \
+  -d '{
+      "memberReferences": ["group:default/example"],
+      "name": "role:default/test",
+      "metadata": { "description": "This is a test role" }
+    }'
+----
+
```
Comment (themr0c) [inline modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc:64]:
```suggestion

.Example request to update a role
[source,subs="+attributes,+quotes"]
----
curl -v -H "Content-Type: application/json" \
  -H "Authorization: Bearer _<token>_" \
  -X PUT "{my-product-url}/api/permission/roles/role/default/test" \
  -d '{
          "oldRole": {
            "memberReferences":  [ "group:default/example" ],
            "name": "role:default/test"
          },
          "newRole": {
            "memberReferences": [ "group:default/example", "user:default/test" ],
            "name": "role:default/test"
          }
        }'
----
```

File: modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+curl -v -H "Content-Type: application/json" \
+  -H "Authorization: Bearer $token" \
+  -X POST "{my-product-url}/api/permission/policies" \
+  -d '[{
+      "entityReference":"role:default/test",
+      "permission": "catalog-entity",
+      "policy": "read", "effect":"allow"
+    }]'
+----
+
```
Comment (themr0c) [inline modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc:77]:
```suggestion

.Example request to create or update a permission policy
[source,subs="+attributes,+quotes"]
----
curl -v -H "Content-Type: application/json" \
  -H "Authorization: Bearer $token" \
  -X POST "{my-product-url}/api/permission/policies/role/default/test" \
  -d '{
         "oldPolicy": [
           { 
             "permission": "catalog-entity", "policy": "read", "effect": "allow"
           }
         ],
         "newPolicy":
           [
             { 
               "permission": "policy-entity", "policy": "read", "effect": "allow"
             }
           ]
       }'
----
```

File: modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+.Example request to create or update a condition
```
Comment (themr0c) [inline modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc:78]:
```suggestion
.Example request to create a condition
```

File: modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+      "resourceType": "catalog-entity",
+      "permissionMapping": ["read"],
+      "conditions": {
+        "rule": "IS_ENTITY_OWNER",
+        "resourceType": "catalog-entity",
+        "params": {"claims": ["group:default/janus-authors"]}
+      }
+    }'
+----
+
```
Comment (themr0c) [inline modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc:98]:
```suggestion

.Example request to create or update a condition
[source,subs="+attributes,+quotes"]
----
curl -v -H "Content-Type: application/json" \
  -H "Authorization: Bearer $token" \
  -X PUT "{my-product-url}/api/permission/roles/conditions/1" \
  -d '{
          "result":"CONDITIONAL",
          "roleEntityRef":"role:default/test",
          "pluginId":"catalog",
          "resourceType":"catalog-entity",
          "permissionMapping": ["read",  "update", "delete"],
          "conditions": {
            "rule": "IS_ENTITY_OWNER",
            "resourceType": "catalog-entity",
            "params": {"claims": ["group:default/janus-authors"]}
          }
       }'
----

```

File: modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc (hunk 8 above)
Comment (AndrienkoAleksandr) [reply modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc:77]: 1) .Example request to create or update a permission policy -> .Example request to update a permission policy
2) POST -> PUT

File: modules/authorization/con-permission-policy-and-role-source.adoc (hunk 1 above)
Comment (themr0c) [reply modules/authorization/con-permission-policy-and-role-source.adoc:22]: This content will need further update. I'll merge it as it is and rework it later, in a dedicated PR.

File: modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+.Procedure
+include::snip-finding-bearer-token.adoc[]
+
+. In a terminal, run the curl command and review the response:
++
+--
+.Default request
+[source,subs="+attributes,+quotes"]
+----
+curl -v -H "Content-Type: application/json" \
```
Comment (themr0c) [inline modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc:18]:
```suggestion
curl -v \
```

File: modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+--
+.Default request
+[source,subs="+attributes,+quotes"]
+----
+curl -v \
+  -H "Authorization: Bearer _<token>_" \
+  -X __<method>__ "{my-product-url}/__<endpoint>__" \
+----
+
+.POST request requiring JSON body data
```
Comment (themr0c) [inline modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc:23]:
```suggestion
.POST or PUT request requiring JSON body data
```

File: modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc [hunk 13]
```diff
@@ -0,0 +1,171 @@
...
+.Prerequisites
+* xref:enabling-and-giving-access-to-rbac[You have access to the RBAC feature].
+
+.Procedure
+include::snip-finding-bearer-token.adoc[]
+
+. In a terminal, run the curl command and review the response:
++
+--
+.Default request
```
Comment (themr0c) [inline modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc:15]:
```suggestion
.GET or DELETE request
```

File: modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc [hunk 14]
```diff
@@ -0,0 +1,171 @@
...
+* `DELETE`: To delete a resource.
+
+pass:c,a,q[{my-product-url}]::
+Enter your {product-short} URL.
+
+__<endpoint>__::
+Enter the xref:ref-rbac-rest-api-endpoints_{context}[API endpoint] to which you want to send a request, such as `/api/permission/policies`.
+
+__<body>__::
+Enter the JSON body with data that your xref:ref-rbac-rest-api-endpoints_{context}[API endpoint] might need with the HTTP `POST` request.
```
Comment (themr0c) [inline modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc:50]:
```suggestion
Enter the JSON body with data that your xref:ref-rbac-rest-api-endpoints_{context}[API endpoint] might need with the HTTP `POST` or `PUT` request.
```

File: modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc [hunk 15]
```diff
@@ -0,0 +1,171 @@
...
+--
+.GET or DELETE request
+[source,subs="+attributes,+quotes"]
+----
+curl -v \
+  -H "Authorization: Bearer _<token>_" \
+  -X __<method>__ "{my-product-url}/__<endpoint>__" \
+----
+
+.POST or PUT request requiring JSON body data
```
Comment (themr0c) [inline modules/authorization/proc-sending-request-to-the-rbac-rest-api-by-using-curl.adoc:23]:
```suggestion
.`GET` or `DELETE` request
[source,subs="+attributes,+quotes"]
----
curl -v \
  -H "Authorization: Bearer _<token>_" \
  -X __<method>__ "{my-product-url}/__<endpoint>__" \
----

.`POST` or `PUT` request requiring JSON body data
```

**Review by pabel-rh — APPROVED** [review]

File: assemblies/assembly-configuring-authorization-in-rhdh.adoc [hunk 16]
```diff
@@ -30,9 +30,15 @@ To apply RBAC in {product-short}:
 include::modules/authorization/proc-enabling-the-rbac-plugin.adoc[leveloffset=+1]
 
 
+include::modules/authorization/con-permission-policy-and-role-source.adoc[leveloffset=+1]
+
+
```
Comment (themr0c) [inline assemblies/assembly-configuring-authorization-in-rhdh.adoc:35]:
```suggestion
```

**Review by gustavolira — APPROVED** [review]

**Pull request merged**